### ISSUE TITLE
Error updating to 2023.4.1 from 2023.4.0

### ISSUE THREAD

**Issue by bmcclure** [bkbilly/lnxlink#56]
**Describe the problem**
2023.4.0 updated successfully and has been running for a while. Today I tried updating to 2023.4.1, and now I'm getting the following error:

```
Traceback (most recent call last):
  File "/home/ben/.local/bin/lnxlink", line 8, in <module>
    sys.exit(main())
  File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
    lnxlink = LNXlink(config_file)
  File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
    self.Addons[addon.service] = addon(self)
  File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
    self.options = self._get_grub_entries()
  File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
    with open('/boot/grub/grub.cfg') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
```

So far I'm not sure how to solve that one, it seems to be related to the new feature added in 2023.4.1 to select a boot option.

Additionally, I was getting the following message:

```
Addon screenshot is not supported, please remove it from your config
No module named 'mss'
```

For that one, I manually ran `pip3 install mss` and then it started working. Shouldn't that be installed as a dependency of lnxlink, or is it expected that it needs to be installed separately?

**Desktop (please complete the following information):**
 - OS: Pop_OS! 22.04
 - Version: 2023.4.1

**Additional context**

Pop_OS! uses Systemd-boot instead of Grub

**Configuration**
Provide the configuration file, remove any sensitive info that might exist:
```yaml
mqtt:
  prefix: lnxlink
  clientId: Dev-Laptop
  statsPrefix: monitor/stats
  server: homeassistant.local
  port: 1883
  auth:
    user: **redacted**
    pass: **redacted**
  discovery:
    enabled: true
  lwt:
    enabled: true
    qos: 1
    retain: true
    connectMsg: 'ON'
    disconnectMsg: 'OFF'
update_interval: 5
```

**Logs**

```txt
Apr 17 12:17:40 pop-os lnxlink[122093]: Traceback (most recent call last):
Apr 17 12:17:40 pop-os lnxlink[122093]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 62, in monitor_run
Apr 17 12:17:40 pop-os lnxlink[122093]:     pub_data = addon.getInfo()
Apr 17 12:17:40 pop-os lnxlink[122093]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/disk_usage.py", line 17, in getInfo
Apr 17 12:17:40 pop-os lnxlink[122093]:     disk_stats = psutil.disk_usage(disk.mountpoint)
Apr 17 12:17:40 pop-os lnxlink[122093]:   File "/usr/lib/python3/dist-packages/psutil/__init__.py", line 2003, in disk_usage
Apr 17 12:17:40 pop-os lnxlink[122093]:     return _psplatform.disk_usage(path)
Apr 17 12:17:40 pop-os lnxlink[122093]:   File "/usr/lib/python3/dist-packages/psutil/_psposix.py", line 170, in disk_usage
Apr 17 12:17:40 pop-os lnxlink[122093]:     st = os.statvfs(path)
Apr 17 12:17:40 pop-os lnxlink[122093]: PermissionError: [Errno 13] Permission denied: '/var/lib/docker/tmp/buildkit-mount879119100'
Apr 18 08:28:34 pop-os systemd[4682]: Stopping LNXLink...
Apr 18 08:28:35 pop-os systemd[4682]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 18 08:28:35 pop-os systemd[4682]: lnxlink.service: Failed with result 'exit-code'.
Apr 18 08:28:35 pop-os systemd[4682]: Stopped LNXLink.
Apr 18 08:28:35 pop-os systemd[4682]: lnxlink.service: Consumed 1d 19h 52min 58.853s CPU time.
-- Boot 4b09bf8358ea4368b81f29800964fffd --
Apr 18 09:43:09 pop-os systemd[4834]: Started LNXLink.
Apr 18 09:43:10 pop-os lnxlink[4845]: Traceback (most recent call last):
Apr 18 09:43:10 pop-os lnxlink[4845]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 62, in monitor_run
Apr 18 09:43:10 pop-os lnxlink[4845]:     pub_data = addon.getInfo()
Apr 18 09:43:10 pop-os lnxlink[4845]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/idle.py", line 16, in getInfo
Apr 18 09:43:10 pop-os lnxlink[4845]:     idle_ms = monitor.get_dbus_idle()
Apr 18 09:43:10 pop-os lnxlink[4845]:   File "/home/ben/.local/lib/python3.10/site-packages/dbus_idle/__init__.py", line 105, in get_dbus_idle
Apr 18 09:43:10 pop-os lnxlink[4845]:     idle_sec = int(stdout.strip())
Apr 18 09:43:10 pop-os lnxlink[4845]: ValueError: invalid literal for int() with base 10: ''
-- Boot 983ce4a82a52487082a0a2029a0a5495 --
Apr 24 09:27:54 pop-os systemd[4804]: Started LNXLink.
Apr 24 09:27:55 pop-os lnxlink[4814]: Traceback (most recent call last):
Apr 24 09:27:55 pop-os lnxlink[4814]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 62, in monitor_run
Apr 24 09:27:55 pop-os lnxlink[4814]:     pub_data = addon.getInfo()
Apr 24 09:27:55 pop-os lnxlink[4814]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/idle.py", line 16, in getInfo
Apr 24 09:27:55 pop-os lnxlink[4814]:     idle_ms = monitor.get_dbus_idle()
Apr 24 09:27:55 pop-os lnxlink[4814]:   File "/home/ben/.local/lib/python3.10/site-packages/dbus_idle/__init__.py", line 105, in get_dbus_idle
Apr 24 09:27:55 pop-os lnxlink[4814]:     idle_sec = int(stdout.strip())
Apr 24 09:27:55 pop-os lnxlink[4814]: ValueError: invalid literal for int() with base 10: ''
Apr 26 11:33:43 pop-os lnxlink[4814]: LNXlink 2023.4.0 started: 3.10.6
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: camera_used
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: send_keys
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: battery
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: idle
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: keep_alive
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: restart
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: sys_updates
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: cpu
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: shutdown
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: suspend
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: screen_onoff
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: required_restart
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: memory
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: media
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: network_download
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: update
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: disk_usage
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: microphone_used
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: xdg_open
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: notify
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: webcam
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: network_upload
Apr 26 11:33:43 pop-os lnxlink[4814]: Successfully loaded addon: bash
Apr 26 11:33:43 pop-os lnxlink[4814]: Can't load addon: idle
Apr 26 11:33:43 pop-os lnxlink[4814]: Connected to MQTT with code 0
Apr 26 11:33:43 pop-os lnxlink[4814]: Connected to MQTT with code 0
Apr 26 11:33:43 pop-os lnxlink[4814]: stopped gracefully
Apr 26 11:33:43 pop-os lnxlink[4814]: Disconnected from MQTT.
Apr 26 11:33:44 pop-os systemd[4804]: lnxlink.service: Consumed 6h 50min 28.781s CPU time.
Apr 26 11:33:44 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 1.
Apr 26 11:33:44 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:44 pop-os systemd[4804]: lnxlink.service: Consumed 6h 50min 28.781s CPU time.
Apr 26 11:33:44 pop-os systemd[4804]: Started LNXLink.
Apr 26 11:33:44 pop-os lnxlink[1311164]: LNXlink 2023.4.1 started: 3.10.6
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: camera_used
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: send_keys
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: battery
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: idle
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: keep_alive
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: restart
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: sys_updates
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: cpu
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: shutdown
Apr 26 11:33:44 pop-os lnxlink[1311164]: Addon screenshot is not supported, please remove it from your config
Apr 26 11:33:44 pop-os lnxlink[1311164]: No module named 'mss'
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: suspend
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: brightness
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: boot_select
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: screen_onoff
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: required_restart
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: memory
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: media
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: network_download
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: update
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: disk_usage
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: microphone_used
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: xdg_open
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: notify
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: webcam
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: network_upload
Apr 26 11:33:44 pop-os lnxlink[1311164]: Successfully loaded addon: bash
Apr 26 11:33:44 pop-os lnxlink[1311164]: Traceback (most recent call last):
Apr 26 11:33:44 pop-os lnxlink[1311164]:   File "/home/ben/.local/bin/lnxlink", line 8, in <module>
Apr 26 11:33:44 pop-os lnxlink[1311164]:     sys.exit(main())
Apr 26 11:33:44 pop-os lnxlink[1311164]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
Apr 26 11:33:44 pop-os lnxlink[1311164]:     lnxlink = LNXlink(config_file)
Apr 26 11:33:44 pop-os lnxlink[1311164]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
Apr 26 11:33:44 pop-os lnxlink[1311164]:     self.Addons[addon.service] = addon(self)
Apr 26 11:33:44 pop-os lnxlink[1311164]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
Apr 26 11:33:44 pop-os lnxlink[1311164]:     self.options = self._get_grub_entries()
Apr 26 11:33:44 pop-os lnxlink[1311164]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
Apr 26 11:33:44 pop-os lnxlink[1311164]:     with open('/boot/grub/grub.cfg') as f:
Apr 26 11:33:44 pop-os lnxlink[1311164]: FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
Apr 26 11:33:44 pop-os systemd[4804]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 26 11:33:44 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:45 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 2.
Apr 26 11:33:45 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:45 pop-os systemd[4804]: Started LNXLink.
Apr 26 11:33:45 pop-os lnxlink[1311169]: LNXlink 2023.4.1 started: 3.10.6
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: camera_used
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: send_keys
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: battery
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: idle
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: keep_alive
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: restart
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: sys_updates
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: cpu
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: shutdown
Apr 26 11:33:45 pop-os lnxlink[1311169]: Addon screenshot is not supported, please remove it from your config
Apr 26 11:33:45 pop-os lnxlink[1311169]: No module named 'mss'
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: suspend
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: brightness
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: boot_select
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: screen_onoff
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: required_restart
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: memory
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: media
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: network_download
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: update
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: disk_usage
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: microphone_used
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: xdg_open
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: notify
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: webcam
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: network_upload
Apr 26 11:33:45 pop-os lnxlink[1311169]: Successfully loaded addon: bash
Apr 26 11:33:45 pop-os lnxlink[1311169]: Traceback (most recent call last):
Apr 26 11:33:45 pop-os lnxlink[1311169]:   File "/home/ben/.local/bin/lnxlink", line 8, in <module>
Apr 26 11:33:45 pop-os lnxlink[1311169]:     sys.exit(main())
Apr 26 11:33:45 pop-os lnxlink[1311169]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
Apr 26 11:33:45 pop-os lnxlink[1311169]:     lnxlink = LNXlink(config_file)
Apr 26 11:33:45 pop-os lnxlink[1311169]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
Apr 26 11:33:45 pop-os lnxlink[1311169]:     self.Addons[addon.service] = addon(self)
Apr 26 11:33:45 pop-os lnxlink[1311169]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
Apr 26 11:33:45 pop-os lnxlink[1311169]:     self.options = self._get_grub_entries()
Apr 26 11:33:45 pop-os lnxlink[1311169]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
Apr 26 11:33:45 pop-os lnxlink[1311169]:     with open('/boot/grub/grub.cfg') as f:
Apr 26 11:33:45 pop-os lnxlink[1311169]: FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
Apr 26 11:33:45 pop-os systemd[4804]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 26 11:33:45 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:45 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 3.
Apr 26 11:33:45 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:45 pop-os systemd[4804]: Started LNXLink.
Apr 26 11:33:46 pop-os lnxlink[1311174]: LNXlink 2023.4.1 started: 3.10.6
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: camera_used
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: send_keys
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: battery
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: idle
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: keep_alive
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: restart
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: sys_updates
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: cpu
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: shutdown
Apr 26 11:33:46 pop-os lnxlink[1311174]: Addon screenshot is not supported, please remove it from your config
Apr 26 11:33:46 pop-os lnxlink[1311174]: No module named 'mss'
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: suspend
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: brightness
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: boot_select
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: screen_onoff
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: required_restart
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: memory
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: media
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: network_download
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: update
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: disk_usage
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: microphone_used
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: xdg_open
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: notify
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: webcam
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: network_upload
Apr 26 11:33:46 pop-os lnxlink[1311174]: Successfully loaded addon: bash
Apr 26 11:33:46 pop-os lnxlink[1311174]: Traceback (most recent call last):
Apr 26 11:33:46 pop-os lnxlink[1311174]:   File "/home/ben/.local/bin/lnxlink", line 8, in <module>
Apr 26 11:33:46 pop-os lnxlink[1311174]:     sys.exit(main())
Apr 26 11:33:46 pop-os lnxlink[1311174]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
Apr 26 11:33:46 pop-os lnxlink[1311174]:     lnxlink = LNXlink(config_file)
Apr 26 11:33:46 pop-os lnxlink[1311174]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
Apr 26 11:33:46 pop-os lnxlink[1311174]:     self.Addons[addon.service] = addon(self)
Apr 26 11:33:46 pop-os lnxlink[1311174]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
Apr 26 11:33:46 pop-os lnxlink[1311174]:     self.options = self._get_grub_entries()
Apr 26 11:33:46 pop-os lnxlink[1311174]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
Apr 26 11:33:46 pop-os lnxlink[1311174]:     with open('/boot/grub/grub.cfg') as f:
Apr 26 11:33:46 pop-os lnxlink[1311174]: FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
Apr 26 11:33:46 pop-os systemd[4804]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 26 11:33:46 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:46 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 4.
Apr 26 11:33:46 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:46 pop-os systemd[4804]: Started LNXLink.
Apr 26 11:33:47 pop-os lnxlink[1311196]: LNXlink 2023.4.1 started: 3.10.6
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: camera_used
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: send_keys
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: battery
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: idle
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: keep_alive
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: restart
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: sys_updates
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: cpu
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: shutdown
Apr 26 11:33:47 pop-os lnxlink[1311196]: Addon screenshot is not supported, please remove it from your config
Apr 26 11:33:47 pop-os lnxlink[1311196]: No module named 'mss'
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: suspend
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: brightness
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: boot_select
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: screen_onoff
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: required_restart
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: memory
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: media
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: network_download
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: update
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: disk_usage
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: microphone_used
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: xdg_open
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: notify
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: webcam
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: network_upload
Apr 26 11:33:47 pop-os lnxlink[1311196]: Successfully loaded addon: bash
Apr 26 11:33:47 pop-os lnxlink[1311196]: Traceback (most recent call last):
Apr 26 11:33:47 pop-os lnxlink[1311196]:   File "/home/ben/.local/bin/lnxlink", line 8, in <module>
Apr 26 11:33:47 pop-os lnxlink[1311196]:     sys.exit(main())
Apr 26 11:33:47 pop-os lnxlink[1311196]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
Apr 26 11:33:47 pop-os lnxlink[1311196]:     lnxlink = LNXlink(config_file)
Apr 26 11:33:47 pop-os lnxlink[1311196]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
Apr 26 11:33:47 pop-os lnxlink[1311196]:     self.Addons[addon.service] = addon(self)
Apr 26 11:33:47 pop-os lnxlink[1311196]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
Apr 26 11:33:47 pop-os lnxlink[1311196]:     self.options = self._get_grub_entries()
Apr 26 11:33:47 pop-os lnxlink[1311196]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
Apr 26 11:33:47 pop-os lnxlink[1311196]:     with open('/boot/grub/grub.cfg') as f:
Apr 26 11:33:47 pop-os lnxlink[1311196]: FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
Apr 26 11:33:47 pop-os systemd[4804]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 26 11:33:47 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:47 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 5.
Apr 26 11:33:47 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:47 pop-os systemd[4804]: Started LNXLink.
Apr 26 11:33:47 pop-os lnxlink[1311201]: LNXlink 2023.4.1 started: 3.10.6
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: camera_used
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: send_keys
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: battery
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: idle
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: keep_alive
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: restart
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: sys_updates
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: cpu
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: shutdown
Apr 26 11:33:47 pop-os lnxlink[1311201]: Addon screenshot is not supported, please remove it from your config
Apr 26 11:33:47 pop-os lnxlink[1311201]: No module named 'mss'
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: suspend
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: brightness
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: boot_select
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: screen_onoff
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: required_restart
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: memory
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: media
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: network_download
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: nvidia_gpu
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: update
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: disk_usage
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: microphone_used
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: xdg_open
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: notify
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: webcam
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: network_upload
Apr 26 11:33:47 pop-os lnxlink[1311201]: Successfully loaded addon: bash
Apr 26 11:33:47 pop-os lnxlink[1311201]: Traceback (most recent call last):
Apr 26 11:33:47 pop-os lnxlink[1311201]:   File "/home/ben/.local/bin/lnxlink", line 8, in <module>
Apr 26 11:33:47 pop-os lnxlink[1311201]:     sys.exit(main())
Apr 26 11:33:47 pop-os lnxlink[1311201]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 292, in main
Apr 26 11:33:47 pop-os lnxlink[1311201]:     lnxlink = LNXlink(config_file)
Apr 26 11:33:47 pop-os lnxlink[1311201]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/__main__.py", line 32, in __init__
Apr 26 11:33:47 pop-os lnxlink[1311201]:     self.Addons[addon.service] = addon(self)
Apr 26 11:33:47 pop-os lnxlink[1311201]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 9, in __init__
Apr 26 11:33:47 pop-os lnxlink[1311201]:     self.options = self._get_grub_entries()
Apr 26 11:33:47 pop-os lnxlink[1311201]:   File "/home/ben/.local/lib/python3.10/site-packages/lnxlink/modules/boot_select.py", line 42, in _get_grub_entries
Apr 26 11:33:47 pop-os lnxlink[1311201]:     with open('/boot/grub/grub.cfg') as f:
Apr 26 11:33:47 pop-os lnxlink[1311201]: FileNotFoundError: [Errno 2] No such file or directory: '/boot/grub/grub.cfg'
Apr 26 11:33:47 pop-os systemd[4804]: lnxlink.service: Main process exited, code=exited, status=1/FAILURE
Apr 26 11:33:47 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:48 pop-os systemd[4804]: lnxlink.service: Scheduled restart job, restart counter is at 6.
Apr 26 11:33:48 pop-os systemd[4804]: Stopped LNXLink.
Apr 26 11:33:48 pop-os systemd[4804]: lnxlink.service: Start request repeated too quickly.
Apr 26 11:33:48 pop-os systemd[4804]: lnxlink.service: Failed with result 'exit-code'.
Apr 26 11:33:48 pop-os systemd[4804]: Failed to start LNXLink.
```

**Comment by bmcclure** [comment]
Quick update about this, I found a workaround for now at least by touching an empty /boot/grub/grub.cfg file. It would be nice not to have to do that though, since my system doesn't use grub.

**Comment by bkbilly** [comment]
This is fixed on version 2023.5.0.
The MSS dependency has been included.
Also it will check all modules for errors with loaded modules. An appropriate message will be displayed on logs.
